src/settings.js

```diff
@@ -488,6 +488,12 @@ var WORKAROUND_IOS_9_RIGHT_SHIFT_BUG = 0;
 // https://stackoverflow.com/questions/54248633/cannot-create-half-float-oes-texture-from-uint16array-on-ipad
 var GL_DISABLE_HALF_FLOAT_EXTENSION_IF_BROKEN = 0;
 
+// Specifies the oldest version of Chrome to target. E.g. a value -s OLDEST_SUPPORTED_CHROME_VERSION=58 would mean that Emscripten generates library code that would
+// contain polyfills and backwards compatibility for Chrome >= 58 and newer. Set this to a small value to target old browsers, and to a high value to reduce code size
+// by dropping polyfill support for older browsers. Default to 0 to mean "all Chrome versions" (although in practice a large number of features require a specific much
+// newer Chrome version, e.g. Wasm or WebGL 2)
+var OLDEST_SUPPORTED_CHROME_VERSION = 0;
```

Comment:
I think perhaps leave tuning the initial defaults for later once all of them have landed, and when `LEGACY_VM_SUPPORT` can migrate to preconfiguring `OLDEST_SUPPORTED_X_VERSION` values.

`MIN_CHROME_VERSION` sounds ok too, I'll rename to that.

Reply:
I think it might be more disruptive to change `MIN_CHROME_VERSION` from 0 to 58 (or such) later, because it kind of changes the actual use and meaning of the flag - from "optionally change this to not support old browsers" to "optionally change this to **yes** support old browsers".

Reply:
I would prefer if the default option were to only support "current" Chrome versions - that is, the last ~3-6 months of Chrome releases (relative to the release date of a version of Emscripten). Realistically, how sure are we even that the rest of Emscripten works on browsers that old? - given, I assume, most developers are not actively testing that far back. And there are very few users on old Chrome versions ([<1%](https://gs.statcounter.com/browser-version-market-share/desktop/worldwide/)).

Maybe this is not in line with the philosophy of the project. I have run across quite a bit of code in Emscripten to handle old browsers, which I would like to remove if it were possible.

Anyway, if you don't want to add that, then defaulting to 58 (now) would be an OK option to me.

Reply:
> Realistically, how sure are we even that the rest of Emscripten works on browsers that old?

I target Samsung Galaxy S2 and Samsung Galaxy Nexus with Chrome 33, iPhone 4S with Safari 8.0, and Raspberry Pi Zero underclocked to 200MHz as the lowest end devices, mostly because they are extremely good performance profiling targets for tiny 2D "Candy Crush" games (and run at 30-60fps). Emulations are needed for `performance.now()`, `requestAnimationFrame()`, fonts, landscape/portrait rotation and especially audio, but things work well on those even.

Most of these tiny playable games are on the order of NES/C64 level of graphical complexity, and if you wanted to run an ad as a playable using asm.js/wasm with Emscripten, you'd really like it to tax only a few % of your browser's gigahertz scale CPU capacity, so it is an useful exercise to look at scaling to ancient browsers.

The only hard limit I have seen so far is on Android 4.3 and older with built-in browser AppleWebKit/534.30 and older, for some reason that completely misparses and misexecutes asm.js code which I was never able to figure out. There are also unfortunate issues with new Chrome and Firefox versions, where sometimes the old built-in browser that ships with a device works better than newest Chrome or Firefox, understandably due to expired lack of testing.